### DISCUSSION TITLE
Configure a prefix any ID's for div/window variables

### DISCUSSION THREAD

**Discussion by cammanderson** [vercel/next.js#43500] [Ideas]
### Describe the feature you'd like to request

Next.js attempts to use a number of specific elements for runtime; `<div id="__next"></div>`, or `__NEXT_DATA__` variables.

These ensure that it isn't possible to have a widget drawn by next.js to be placed on an alternative website, as there will be potential conflict with another next.js application that uses the same ID's.

### Describe the solution you'd like

Add a configuration option to next.config.ts that could allow developers to change or prefix these global keys.

Have control over triggering the initialisation function that would use ReactDOM to hydrate from a specific selector that contains the HTML.

### Describe alternatives you've considered

Alternatives are limited, but could be;
- Custom server that would be capable of rendering the components; various components use `next` packages and could have problem rendering these; essentially trying to make use of the components and maybe some of the internals of `next` 
- Not implementing a widget in next.js

**Comment by Fredkiss3** [comment]
For rendering, normally you would use react and not tie your widget to nextjs right ?
Because if it's tied to nextjs you could only use it in a next app or maybe with nextjs server running.

And if you use react for your widget, you can create a library and control when and where to render your widget.

**Comment by pablosantos22** [comment]
Hello!
move forward on this topic?
In the company I work for, we use micro-frontends and we need to create a preload where the static files would already be ready to be rendered inside a div with an id related to the micro-frontend so that it didn't cause transmission. I haven't found a way to customize the id of this root div. The idea would be to have id=app_1, id=app_2 etc..
